cpp/tests/pagerank/mg_pagerank_test.cpp

```diff
@@ -315,8 +310,38 @@ TEST_P(Tests_MGPageRank_File, CheckInt32Int32FloatFloat)
 
 TEST_P(Tests_MGPageRank_Rmat, CheckInt32Int32FloatFloat)
 {
-  auto param = GetParam();
-  run_current_test<int32_t, int32_t, float, float>(std::get<0>(param), std::get<1>(param));
+  auto param            = GetParam();
+  auto pagerank_usecase = std::get<0>(param);
```

Comment:
This approach will override *all* rmat tests, right?  In this case we have a small rmat use case that has the correctness test enabled and a large rmat use case that has the correctness disabled.  I believe this change would consequently run the test twice for each combination (32/32, 32/64, 64/64) once with and once without validation.  Not sure that's what we want.

Reply:
Perhaps we create a separate test to address this.  I'm bad at naming things, so imagine better names:

1. Create a test suite called Benchmark_MGPageRank_Rmat.  Disable validation in that test suite and provide a single rmat test (something small so that when we're doing unit tests in CI it won't take long)
2. Add this logic to *that* unit test, but not the basic unit tests
3. When you want to run with these overriding parameters you can run:  `mpirun -np 2 tests/MG_PAGERANK_TEST --gtest_filter=Benchmark_MGPageRank_Rmat*` and append whatever overriding parameters are desired

That would allow you to get a clean run without the extra runs.

Reply:
> This approach will override _all_ rmat tests, right?

Yes, without ```--gtest_filter```, no, with the filter. This was intended to use in the benchmarking mode with something like ```--gtest_filter=rmat_large_tests/Tests_MGPageRank_Rmat.CheckInt32Int32*```.

Reply:
See the "Exemplar benchmark scripts" part of this PR description.

Reply:
With the filter to run only rmat_large_tests, I think the current code can achieve what we can do this Benchmark_MGPageRank_Rmat without actually adding additional code for that (and with minimum interruption in our C++ testing work process).

Reply:
OK.  So I should read *all* of the description when reviewing the PR, just not jump to the code :-)

It would be nice to encapsulate this logic somehow rather than having to repeat it in every test.  Could we create a function that updates the rmat_usecase that can be reusable across all of the tests (SG or MG)?  Perhaps it could be a method in the rmat_usecase class?

Reply:
Also, did we want to allow for specifying other input files (an analog for the file_usecase)?  That *might* require some refactoring of the tests since most of our tests don't have a single file test without validation.

Reply:
> It would be nice to encapsulate this logic somehow rather than having to repeat it in every test.

Agreed, I will push this update in the next commit.

> Also, did we want to allow for specifying other input files (an analog for the file_usecase)?

Yeah... eventually, but let's think about this later (I guess we may need to add something like file_large_tests... but let's worry about this when we have a specific usecase).